### PR TITLE
Added 404 page to site

### PR DESCRIPTION
**Pull request by andrewstech**
## Proposed change

Added 404 page

## Type of change

Added 404 page  for broken links

## Additional information

I've added a custom 404 page for the home-assistant website.

## Checklist

- [ x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Comment by homeassistant** [comment]
Hello @andrewstech,

When attempting to inspect the commits of your pull request for CLA signature status among all authors we encountered commit(s) which were not linked to a GitHub account, thus not allowing us to determine their status(es).

The commits that are missing a linked GitHub account are the following:

- [`ad7f000c8ef91a81f3259061205a704033f799b9`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant.io&#x2F;pull&#x2F;16496&#x2F;commits&#x2F;ad7f000c8ef91a81f3259061205a704033f799b9) - This commit has something that looks like an email address (info@andrewstech.me). Maybe try linking that to GitHub?.
- [`65efd764135538028d4385bbf809385eff99f2e9`](https:&#x2F;&#x2F;github.com&#x2F;home-assistant&#x2F;home-assistant.io&#x2F;pull&#x2F;16496&#x2F;commits&#x2F;65efd764135538028d4385bbf809385eff99f2e9) - This commit has something that looks like an email address (info@andrewstech.me). Maybe try linking that to GitHub?.

Unfortunately, **we are unable to accept this pull request until this situation is corrected.**

Here are your options:

1. If you had an email address set for the commit that simply wasn't linked to your GitHub account you can link that email now and it will retroactively apply to your commits. The simplest way to do this is to click the link to one of the above commits and look for a blue question mark in a blue circle in the top left. Hovering over that bubble will show you what email address you used. Clicking on that button will take you to your email address settings on GitHub. Just add the email address on that page and you're all set. GitHub has more information about this option [in their help center](https://help.github.com/articles/setting-your-email-in-git/#commits-on-github-arent-linking-to-my-account).

2. If you didn't use an email address at all, it was an invalid email, or it's one you can't link to your GitHub, you will need to change the authorship information of the commit and your global Git settings so this doesn't happen again going forward. GitHub provides some great instructions on how to change your authorship information [in their help center](https://help.github.com/articles/setting-your-email-in-git/).
   * If you only made a single commit you should be able to run
      ```
      git commit --amend --author="Author Name <email@address.com>"
      ```
      (substituting Author Name and email@address.com for your actual information) to set the authorship information.
   * If you made more than one commit and the commit with the missing authorship information is not the most recent one you have two options:
       1. You can re-create all commits missing authorship information. This is going to be the easiest solution for developers that aren't extremely confident in their Git and command line skills.
       2. You can use [this script](https://help.github.com/articles/changing-author-info/) that GitHub provides to rewrite history. **Please note:** this should be used only if you are very confident in your abilities and understand its impacts.
   * Whichever method you choose, I will come by to re-check the pull request once you push the fixes to this branch.

We apologize for this inconvenience, especially since it usually bites new contributors to Home Assistant. We hope you understand the need for us to protect ourselves and the great community we all have built legally. The best thing to come out of this is that you only need to fix this once and it benefits the entire Home Assistant and GitHub community.

Thanks, I look forward to checking this PR again soon! :heart:

**Review by andrewstech — COMMENTED** [review]
git email fixed

**Comment by homeassistant** [comment]
Hi @andrewstech, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;home-assistant.io#16496).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: package.json [hunk 1]
```diff
@@ -2,7 +2,10 @@
   "name": "home-assistant.io",
   "description": "Home Assistant Website & Documentation",
   "version": "1.0.0",
-  "dependencies": {},
+  "dependencies": {
+    "bundler": "^0.8.0",
+    "gem": "^2.4.3"
```
Comment (frenck) [inline package.json]: Why are these in here?

File: Gemfile.lock [hunk 2]
```diff
@@ -138,9 +139,10 @@ DEPENDENCIES
...
   tzinfo (~> 2.0)
   tzinfo-data
+  wdm (~> 0.1.0)
 
 RUBY VERSION
    ruby 2.6.2p47
 
 BUNDLED WITH
-   2.0.1
+   2.2.9
```
Comment (frenck) [inline Gemfile.lock]: We cannot use a different bundler version.

File: Gemfile.lock [hunk 3]
```diff
@@ -138,9 +139,10 @@ DEPENDENCIES
   stringex (= 2.8.5)
   tzinfo (~> 2.0)
   tzinfo-data
+  wdm (~> 0.1.0)
```
Comment (frenck) [inline Gemfile.lock]: ?

File: 404.html [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    padding-top: 110px;
+  }
+}
+</style>
+
+	<!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
+	<!-- WARNING: Respond.js doesn't work if you view the page via file:// -->
+	<!--[if lt IE 9]>
+		  <script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
+		  <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
```
Comment (frenck) [inline 404.html]: Why do we need these?

File: 404.html [hunk 5]
```diff
@@ -0,0 +1,142 @@
...
+  }
+  .notfound {
+    padding-left: 15px;
+    padding-right: 15px;
+    padding-top: 110px;
+  }
+}
+</style>
+
+	<!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
```
Comment (frenck) [inline 404.html]: Please don't use tabs.

File: source/404.html [hunk 6]
```diff
@@ -0,0 +1,142 @@
+<!DOCTYPE html>
+<html lang="en">
```
Comment (frenck) [inline source/404.html]: Why another 404?

**Review by frenck — CHANGES_REQUESTED** [review]
Some comments to look at.

I'm not sure how Netlify would pick up this page? Could you elaborate on that?

File: package.json (hunk 1 above)
Comment (andrewstech) [reply package.json]: > Some comments to look at.
> 
> I'm not sure how Netlify would pick up this page? Could you elaborate on that?

Not sure I believe Gem has changed these files. I will change them back

File: 404.html (hunk 4 above)
Comment (andrewstech) [reply 404.html]: They have been removed

File: source/404.html (hunk 6 above)
Comment (andrewstech) [reply source/404.html]: Removed duplicated file

File: source/404.html [hunk 7]
```diff
@@ -0,0 +1,132 @@
...
+}
+
+.notfound .notfound-404 {
+  position: absolute;
+  left: 0;
+  top: 0;
+  display: inline-block;
+  width: 140px;
+  height: 140px;
+  background-image: url('https://upload.wikimedia.org/wikipedia/commons/thumb/6/6e/Home_Assistant_Logo.svg/1200px-Home_Assistant_Logo.svg.png');
```
Comment (frenck) [inline source/404.html]: Lets make sure the logo is hosted from our own website 😉

File: 404.html (hunk 5 above)
Comment (andrewstech) [reply 404.html]: Tabs removed

File: source/404.html (hunk 7 above)
Comment (andrewstech) [reply source/404.html]: Sure I do that now

Comment (andrewstech) [reply source/404.html]: Done pushing changes now.

File: source/404.html [hunk 8]
```diff
@@ -0,0 +1,132 @@
...
+<div class="notfound">
+<div class="notfound-404"></div>
+<h1>404</h1>
+<h2>Oops! Page Not Be Found</h2>
+<p>Sorry but the page you are looking for does not exist, have been removed. name changed or is temporarily unavailable.</p>
+<a href="https://home-assistant.io/">Back to homepage</a>
+</div>
+</div>
+
+</body><!-- This templates was made by Colorlib (https://colorlib.com) -->
```
Comment (frenck) [inline source/404.html]: Are we allowed to use this template? Is the license matching for our project?

Comment (andrewstech) [reply source/404.html]: It says

You are allowed to use templates for personal and business usage.
You are allowed to customize templates however you like.
You are allowed to convert/port HTML templates for use for any CMS.
You are allowed to share and distribute them under Colorlib brand name.

This is from there site https://colorlib.com/wp/licence/

Comment (andrewstech) [reply source/404.html]: As far as I can tell yes we can use it.

Comment (frenck) [reply source/404.html]: Sure, but the copyright needs to stay intact. As in, this is not a fully open-source template, as it comes with additional licensing (that btw doesn't match the license they pretend to have).

Nevertheless, a simple error in a contribution, later on, can cause problems. Additionally, I don't think this matches our CLA (which you have signed for this PR): <https://www.home-assistant.io/developers/cla/>

Alternatively, we could create a 404.markdown file in the root and render that in our own, existing, layout instead?

**Comment by andrewstech** [comment]
> Some comments to look at.
> 
> I'm not sure how Netlify would pick up this page? Could you elaborate on that?

Jekyll will set the 404 page automatically to the file 404.html 

You can see an example of this, if you use the the [deploy preview](https://deploy-preview-16496--home-assistant-docs.netlify.app/)  go to an non existing page and it should work. [For example](https://deploy-preview-16496--home-assistant-docs.netlify.app/nothinghere)

File: source/404.html (hunk 8 above)
Comment (andrewstech) [reply source/404.html]: Sure I will have a playaround with markdown.  I will see what I can come up with.

Comment (andrewstech) [reply source/404.html]: Done.

**Review by frenck — APPROVED** [review]
Thanks, @andrewstech 👍

**Pull request merged**